In [ ]:
!pip install katna

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.7/62.7 kB 3.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for ffmpy: filename=ffmpy-0.3.0-py3-none-any.whl size=4707 sha256=1efc077f5f7a1cce93bcdd088778c95edc00115f960a3d77bdcb8ba22c4fcced
  Stored in directory: /root/.cache/pip/wheels/91/e2/96/f676aa08bfd789328c6576cd0f1fde4a3d686703bb0c247697
Successfully built ffmpy


In [ ]:
from Katna.image import Image

In [ ]:
img_module = Image()

Caffe Model file downloaded.
Prototext file downloaded.
Model file downloaded.


In [ ]:
import os.path
import cv2
from Katna.image import Image
from Katna.writer import ImageCropDiskWriter

In [ ]:
import ntpath

In [ ]:
input_dir_path = '/content/drive/MyDrive/Embedded AI Project /Test/files'

crop_list = img_module.resize_image_from_dir(
   dir_path=input_dir_path,
   target_width=96,
   target_height=96,
   down_sample_factor=8
)

In [ ]:
import os.path
from Katna.image import Image
import os
import ntpath


def main():

    img_module = Image()

    # folder to save resized images
    output_folder_resized_image = '/content/drive/MyDrive/Embedded AI Project /Test/files'#"resizedimages"

    if not os.path.isdir(os.path.join(".", output_folder_resized_image)):
        os.mkdir(os.path.join(".", output_folder_resized_image))

    # resized image dimensions
    resize_width = 96
    resize_height = 96

    # Input folder file path
    # input_folder_path = os.path.join(".", "tests", "data")
    # print(f"input_folder_path = {input_folder_path}")

    resized_images = img_module.resize_image_from_dir(
        dir_path='/content/drive/MyDrive/Embedded AI Project /Test/files',
        target_width=resize_width,
        target_height=resize_height,
        down_sample_factor=8,
    )

    for filepath, resized_image in resized_images.items():
        # name of the image file
        filename = ntpath.basename(filepath)
        name = filename.split(".")[0]
        # folder path where the images will be stored
        img_module.save_image_to_disk(
            resized_image, output_folder_resized_image, name + "_resized" + "_", ".jpeg"
        )


main()

In [ ]:
len(labels)

496

In [ ]:
y = np.array(data)
y.shape

(496, 224, 224, 3)

In [ ]:
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /Images_of_Waste/YOLO_imgs"

data = []
labels = []

for filename in os.listdir(dataset_dir):
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        label_filename = os.path.splitext(filename)[0] + ".txt"
        with open(os.path.join(dataset_dir, label_filename)) as f:
            label = int(f.readline().strip()[0])
        data.append(image)
        labels.append(label)

data = np.array(data, dtype="float32")
labels = np.array(labels)

data /= 255.0

In [ ]:
# data

In [ ]:
a = np.array(data)
b = np.array(data)

In [ ]:
c = np.concatenate([a, b])

In [ ]:
c.shape

(992, 224, 224, 3)

###DATA from rawimgs

In [ ]:
# AluCan
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /newimages/alucan"

data_alu = []
labels_alu = []
count = 0
for filename in os.listdir(dataset_dir):
    # if count == 800:
    #   break
    
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_alu.append(image)
        labels_alu.append(0)
        count +=1

data_alu = np.array(data_alu, dtype="float32")
labels_alu = np.array(labels_alu)

data_alu /= 255.0

In [ ]:
print('data', data_alu.shape)
print('labels', labels_alu.shape)

data (457, 224, 224, 3)
labels (457,)


In [ ]:
#Glass
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /newimages/glass"

data_glass = []
labels_glass = []
gcount = 0
for filename in os.listdir(dataset_dir):
    # if gcount == 800:
    #   break

    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_glass.append(image)
        labels_glass.append(1)
        gcount += 1

data_glass = np.array(data_glass, dtype="float32")
labels_glass = np.array(labels_glass)

data_glass /= 255.0

In [ ]:
print('data', data_glass.shape)
print('labels', labels_glass.shape)

data (468, 224, 224, 3)
labels (468,)


In [ ]:
#HDPEM
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /newimages/hdpem"

data_hdpem = []
labels_hdpem = []
hcount = 0
for filename in os.listdir(dataset_dir):
    # if hcount == 800:
    #   break
    
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_hdpem.append(image)
        labels_hdpem.append(2)
        hcount+=1 

data_hdpem = np.array(data_hdpem, dtype="float32")
labels_hdpem = np.array(labels_hdpem)

data_hdpem /= 255.0


In [ ]:
print('data', data_hdpem.shape)
print('labels', labels_hdpem.shape)

data (173, 224, 224, 3)
labels (173,)


In [ ]:
#PET
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /newimages/pet"

data_pet = []
labels_pet = []
pcount = 0
for filename in os.listdir(dataset_dir):
    # if pcount == 800:
    #   break
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_pet.append(image)
        labels_pet.append(3)
        pcount+=1

data_pet = np.array(data_pet, dtype="float32")
labels_pet = np.array(labels_pet)

data_pet /= 255.0

In [ ]:
print('data', data_pet.shape)
print('labels', labels_pet.shape)

data (220, 224, 224, 3)
labels (220,)


Below, we load added images from second set of data samples that we collected:

****

Appending our images and corresponding labels to existing data:

In [ ]:
# Our AluCan (oalu)
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /ourimages/alucan"

data_oalu = []
labels_oalu = []
count = 0
for filename in os.listdir(dataset_dir):
    # if count == 800:
    #   break
    
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_oalu.append(image)
        labels_oalu.append(0)
        count +=1

data_oalu = np.array(data_oalu, dtype="float32")
labels_oalu = np.array(labels_oalu)

data_oalu /= 255.0

In [ ]:
print('data', data_oalu.shape)
print('labels', labels_oalu.shape)

data (236, 224, 224, 3)
labels (236,)


In [ ]:
#Our Glass
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /ourimages/glass"

data_oglass = []
labels_oglass = []
gcount = 0
for filename in os.listdir(dataset_dir):
    # if gcount == 800:
    #   break

    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_oglass.append(image)
        labels_oglass.append(1)
        gcount += 1

data_oglass = np.array(data_oglass, dtype="float32")
labels_oglass = np.array(labels_oglass)

data_oglass /= 255.0

In [ ]:
print('data', data_oglass.shape)
print('labels', labels_oglass.shape)

data (246, 224, 224, 3)
labels (246,)


In [ ]:
#Our HDPEM
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /ourimages/hdpem"

data_ohdpem = []
labels_ohdpem = []
hcount = 0
for filename in os.listdir(dataset_dir):
    # if hcount == 800:
    #   break
    
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_ohdpem.append(image)
        labels_ohdpem.append(2)
        hcount+=1 

data_ohdpem = np.array(data_ohdpem, dtype="float32")
labels_ohdpem = np.array(labels_ohdpem)

data_ohdpem /= 255.0

In [ ]:
print('data', data_ohdpem.shape)
print('labels', labels_ohdpem.shape)

data (274, 224, 224, 3)
labels (274,)


In [ ]:
#Our PET
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /ourimages/pet"

data_opet = []
labels_opet = []
pcount = 0
for filename in os.listdir(dataset_dir):
    # if pcount == 800:
    #   break
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_opet.append(image)
        labels_opet.append(3)
        pcount+=1

data_opet = np.array(data_opet, dtype="float32")
labels_opet = np.array(labels_opet)

data_opet /= 255.0

In [ ]:
print('data', data_opet.shape)
print('labels', labels_opet.shape)

data (237, 224, 224, 3)
labels (237,)


In [ ]:
#Non-trash Data
import os
import cv2
import numpy as np

dataset_dir = "/content/drive/MyDrive/Embedded AI Project /ourimages/nontrash"

data_nontrash = []
labels_nontrash = []
pcount = 0
for filename in os.listdir(dataset_dir):
    # if pcount == 800:
    #   break
    if filename.endswith(".jpg"):
        # read the image and resize it to a fixed size
        image = cv2.imread(os.path.join(dataset_dir, filename))
        image = cv2.resize(image, (224, 224))

        # read the label from the corresponding .txt file
        # label_filename = os.path.splitext(filename)[0] + ".txt"
        # with open(os.path.join(dataset_dir, label_filename)) as f:
        #     label = int(f.readline().strip()[0])
        data_nontrash.append(image)
        labels_nontrash.append(4)
        pcount+=1

data_nontrash = np.array(data_nontrash, dtype="float32")
labels_nontrash = np.array(labels_nontrash)

data_pet /= 255.0

In [ ]:
print('data', data_nontrash.shape)
print('labels', labels_nontrash.shape)

data (716, 224, 224, 3)
labels (716,)


Concatenating Data from all 4 classes into one dataset:

TOTAL Dataset count: (From images above and below, this cell)

Aluminium Cans (Class 0): 693

Glass (Class 1): 714

HDPEM (Class 2): 457

PET (Class 3): 447

Total: 2311

In [ ]:
data_iow = np.concatenate([data_alu, data_glass, data_hdpem, data_pet, data_oalu, data_oglass, data_ohdpem, data_opet])

In [ ]:
data_iow.shape

(2311, 224, 224, 3)

Concatenating labels from all 4 classes into one dataset:

In [ ]:
labels_iow = np.concatenate([labels_alu, labels_glass, labels_hdpem, labels_pet, labels_oalu, labels_oglass, labels_ohdpem, labels_opet])

In [ ]:
labels_iow.shape

(2311,)

Saving Data (Images) and corresponding Labels as .npy files in 'data_iow' file in Project Drive:

In [ ]:
import os
import numpy as np

home_dir = os.path.expanduser("~")
data_path = '/content/drive/MyDrive/Embedded AI Project /data_iow/CombinedData'

if not os.path.exists(data_path):
    os.makedirs(data_path)

np.save(os.path.join(data_path, "data.npy"), data_iow)
np.save(os.path.join(data_path, "labels.npy"), labels_iow)